XIPs/xip-31-message-backup-request-system.md

```diff
@@ -0,0 +1,466 @@
+
+There are a number of cases where message history synchronization is desirable:
+
+- User installs App A and starts engaging in conversations. Later, they install App B on the same device and want access to their existing conversations and messages.
+- User installs App A and starts engaging in conversations. Later, they install App B (or another copy of App A) on a second device and desire access to their existing conversations and messages.
+- User installs App A and starts engaging in conversations. They lose access to the device App A was installed on and purchase a new device. They then need to restore their previous conversations and messages to the new device.
+- User installs App A and starts engaging in conversations. App A is later compromised, and they need to migrate their messages and conversations to a different application.
+
+It's also worth noting that there are many use-cases for XMTP where message history synchronization is not necessary or desirable.
+
+- A bot that replies to messages from users. Many bots don't care about messages sent before they were created.
+- A service that sends notifications to a list of subscribers via XMTP.
+- An application that allows users to talk to customer support only needs access to the newly created conversation with the support agent.
+- A marketplace application that wants to allow buyers and sellers to communicate about a particular item. These would only be new conversations and may not care about pre-existing messages.
+
+### Design goals
+
+1. Message history synchronization should not compromise the Forward Secrecy or Post Compromise Security properties of XMTP V3. We expect the majority of consumer applications using XMTP to have some form of message history syncing enabled.
+1. Message history synchronization should be opt-in for developers.
+1. It should be simple for developers to implement message history synchronization correctly. Implementation work should be focused on application-specific UX choices, with the SDK responsible for core transport and security decisions.
+1. Compatibility between applications is expected. Applications should be able to share history irrespective of operating system, SDK version, or device type (web vs mobile vs server).
+1. Even if developers of some popular applications choose to not support message history synchronization, it should be simple for users to opt enable themselves.
+
+## Specification
+
+This specification offers the user two modes of message backup, which accomodate different use-cases: Remote Message Backups and Backup Account Files. Remote Message Backups are the lowest friction solution, provided the blockchain account has access to another existing installation. Backup Account Files are an emergency solution for cases where the user has lost access to all of their XMTP applications (for example, if they lost the only device they have used to connect to XMTP).
```

Comment:
I may be splitting hairs, but some social media users have preconceptions about the term "backup account". Suggest we consider naming our 2nd use case "Account Backup File" to avoid concept collision with the above.

Reply:
I worry "Account Backup File" might lead developers astray. This process is creating a secondary XMTP account just for backups, and then saving it to a file. When I hear "Account Backup File" it's easy to assume that it is a backup of your regular account.

We can workshop the name a bit. I'm not married to "Backup Account File".